PhysiCell/custom_modules/custom.cpp

```diff
@@ -275,9 +275,141 @@ void naive_bcell_phenotype( Cell* pCell, Phenotype& phenotype , double dt ) {
         // let's say, at max pressure 98% of the cells should enter apoptosis within 60 [1/min]
         // 60[min] * rM[1/min] = 1
         double s1Apoptosis = 0.98 / 60;
-
 	// rule of pressure (future: and alignment score) to steer apoptosis rate
 	double rApoptosis = s0Apoptosis + (s1Apoptosis - s0Apoptosis) * rPressure;
-        set_single_behavior( pCell, "apoptosis" , rApoptosis );
+        //set_single_behavior( pCell, "apoptosis" , rApoptosis );
         printf("apoptosis min: {%g}\tmax: {%g}\tset: {%g}\n", s0Apoptosis, s1Apoptosis, rApoptosis);
+
+	///Jays code:
+	///we need to tranform when the "correct" sequence is recieved
+	///first lets always transform!
+	//boo
+	double isTransform = alignment(antibodySequence, antigenSequence);
+	printf("score: %g", isTransform);
+	long antibodyLen = antibodySequence.value.size();
+	long antigenLen = antigenSequence.value.size();
+	double anti;
+	
+	if(isTransform >= .7) {
+		set_single_behavior(pCell, "transform to B_follicular", 9e9);
+	}
+	else if(isTransform >= .90){
+		set_single_behavior(pCell, "transform to B_plasma", 9e9);
+	}
+	else{
+		printf("There was no change, stay naive\n");
+	}
```

Comment:
Do you know for certain whether all the data is kept after a transform like the B cell's antigen and antibody?

Reply:
I think it's important because a plasma cell is only a master of killing one type of invader.

Reply:
this i s already merged into the hamming branch. i close this pull request.
thank you,  Jay!

Elmar